### ISSUE TITLE
[3.x]: searchScore below 1 are set to `null`

### ISSUE THREAD

**Issue by dgsiegel** [craftcms/cms#11973]
### Description

Suppose you have a simple entry search like the following:

```
{% set results = craft.entries()
  .search('foo')
  .orderBy('score')
  .all() %}

{% for r in results %}
  {{ r.searchScore }}
{% endfor %}

```
This will output all search scores of matching entries. However all numbers were cast to an integer. 

If you inspect the search scores at this point (https://github.com/craftcms/cms/blob/v3/src/elements/db/ElementQuery.php#L2909), you'll get the actual floating numbers.

At some point however they are cast to an integer, and I think the culprit could be this line (https://github.com/craftcms/cms/blob/v3/src/base/Element.php#L1837):

`        $this->searchScore = (int)$this->searchScore ?: null;`

### Expected behavior

`searchScore`'s below 1 should be set to a 0 or the actual floating value.

### Actual behavior

`searchScore`'s below 1 are set to `null` instead of a number.

### Craft CMS version

3.7.54

### PHP version

PHP 8.0.18

### Database type and version

MySQL

### Installed plugins and versions

-

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just released Craft 3.7.55 and 4.2.5 with a fix.

**Comment by dgsiegel** [comment]
Awesome, thanks!
